src/NUnitFramework/framework/Internal/StringTokenEnumerator.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace NUnit.Framework.Internal
+{
+    /// <summary>
+    /// Allows enumerating sub-strings from a string based on separator char. Will not return anything for empty string!
+    /// Basically a less allocating version of string.Split.
+    /// </summary>
+    internal struct StringTokenEnumerator : IEnumerator<string>, IEnumerable
+    {
+        private int _currentStartIndex;
+        private readonly string _target;
+        private readonly char _separator;
+
+        /// <summary>
+        /// Constructs a new enumerator against given target with given separator.
+        /// </summary>
+        /// <param name="target">Target string to enumerate.</param>
+        /// <param name="separator">Separator between tokens to return.</param>
+        public StringTokenEnumerator(string target, char separator)
+        {
+            _currentStartIndex = 0;
+            _target = target;
+            _separator = separator;
+            Current = null;
+            Current = null;
+        }
+
+        /// <inheritdoc />
+        public bool MoveNext()
+        {
+            int start = _currentStartIndex;
+
+            if (start >= _target.Length)
+            {
+                return false;
+            }
+
+            var endIndex = _target.IndexOf(_separator, start);
+            if (endIndex == -1)
+            {
+                Current = _target.Substring(start);
+                _currentStartIndex = _target.Length;
+            }
+            else
+            {
+                Current = _target.Substring(start, endIndex - start);
+                _currentStartIndex = endIndex + 1;
+            }
+
+            return true;
+        }
+
+        /// <inheritdoc />
+        public void Reset()
+        {
+            _currentStartIndex = -2;
```

Comment:
This seem wrong. After a call to `Reset` the following calls to `MoveNext` will fail as we cannot have a negative start. I think we either should return `0` instead of `-2` or perhaps throw `NotSupportedException` - I prefer to return `0`.

From the documentation https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerator.reset?view=net-6.0
> The [Reset](https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerator.reset?view=net-6.0) method is provided for COM interoperability. It does not necessarily need to be implemented; instead, the implementer can simply throw a [NotSupportedException](https://docs.microsoft.com/en-us/dotnet/api/system.notsupportedexception?view=net-6.0).

Reply:
Ah, yes, I did some expirements and this was left. Thanks for checking for requirements, will go with NotSupportedException.